bindata/network/multus/multus.yaml

```diff
@@ -29,101 +108,204 @@ spec:
+        image: {{.MultusImage}}
+        command: ["/entrypoint/fipsbincopy.sh"]
+        volumeMounts:
+        - mountPath: /entrypoint
+          name: fips-binary-copy
+        - mountPath: /host/opt/cni/bin
+          name: cnibin
+        - mountPath: /host/etc/os-release
+          name: os-release
+          readOnly: true
+        env:
+        - name: RHEL7_SOURCE_DIRECTORY
+          value: "/usr/src/multus-cni/rhel7/bin/"
+        - name: RHEL8_SOURCE_DIRECTORY
+          value: "/usr/src/multus-cni/rhel8/bin/"
+        - name: DEFAULT_SOURCE_DIRECTORY
+          value: "/usr/src/multus-cni/bin/"
+        - name: DESTINATION_DIRECTORY
+          value: "/host/opt/cni/bin/"
       - name: cni-plugins
         image: {{.CNIPluginsImage}}
-        command: 
-          - /bin/bash
-          - -c
-          - |
-            #!/bin/bash
-            set -x
-            . /host/etc/os-release
-            rhelmajor=
-            # detect which version we're using in order to copy the proper binaries
-            case "${ID}" in
-              rhcos) rhelmajor=8
-              ;;
-              rhel) rhelmajor=$(echo "${VERSION_ID}" | cut -f 1 -d .) 
-              ;;
-              fedora)
-                if [ "${VARIANT_ID}" == "coreos" ]; then
-                  rhelmajor=8
-                else 
-                  echo "FATAL ERROR: Unsupported Fedora variant=${VARIANT_ID}"
-                  exit 1
-                fi
-              ;;
-              *) echo "FATAL ERROR: Unsupported OS ID=${ID}"; exit 1
-              ;;
-            esac
-            # Set which directory we'll copy from, detect if it exists
-            # When it doesn't exist, fall back to the original directory.
-            SOURCEDIR=/usr/src/plugins/rhel${rhelmajor}/bin/
-            if [ -d "${SOURCEDIR}" ]; then
-              echo "Detected OS version ${rhelmajor}, ${SOURCEDIR} exists"
-            else
-              echo "Source directory unavailable for OS version: ${rhelmajor}"
-              SOURCEDIR=/usr/src/plugins/bin/
-            fi
-            cp -rf ${SOURCEDIR}* /host/opt/cni/bin
-
-        securityContext:
-          privileged: true
+        command: ["/entrypoint/fipsbincopy.sh"]
         volumeMounts:
+        - mountPath: /entrypoint
+          name: fips-binary-copy
         - mountPath: /host/opt/cni/bin
           name: cnibin
         - mountPath: /host/etc/os-release
           name: os-release
           readOnly: true
-      - name: multus-binary-copy
-        image: {{.MultusImage}}
+        env:
+        - name: RHEL7_SOURCE_DIRECTORY
+          value: "/usr/src/plugins/rhel7/bin/"
+        - name: RHEL8_SOURCE_DIRECTORY
+          value: "/usr/src/plugins/rhel8/bin/"
+        - name: DEFAULT_SOURCE_DIRECTORY
+          value: "/usr/src/plugins/bin/"
+        - name: DESTINATION_DIRECTORY
+          value: "/host/opt/cni/bin/"
+      - name: routeoverride-cni
+        image: {{.RouteOverrideImage}}
```

Comment:
Do we really need to install this onto every node in every cluster? Can't we make this conditional on configuration somehow? (ie, it only gets deployed if you're using it). Similarly, does this really belong in the multus daemonset? It's really more "additional-networks" than "multus" isn't it?

Alternatively, perhaps we could copy all of the CNI binaries into a single image...

Reply:
Good considerations all. As for conditional copy, the only thing that really works this way is for DHCP CNI -- and in that case it's because it needs additional components (like the DHCP CNI running in daemon mode).

Previously, all the CNI binaries have been copied in these init containers for Multus... Well, the reference CNI plugins + the Multus binary, and now we add two more. I'll look at moving the CNI binary copies to be organized in the additional networks header...

I like the idea of condensing all of the CNI binaries into a single image, I think going forward that would likely be cleaner. However, there's a variety of upstream sources for these which adds some degree of complexity.